### DISCUSSION TITLE
Auto Restart Shizuku when network is changed.

### DISCUSSION THREAD

**Discussion by Eshanatnight** [RikkaApps/Shizuku#322] [Q&A]
I wanted to ask if there is any way to auto restart Shizuku, when the network is changed. Say From WiFi to Mobile Data

**Comment by RikkaW** [comment]
On my Pixel 5 (unrooted with Android 13 Jul patch), changing the network will not cause Shizuku to stop. 

Have you followed all the steps listed here? https://shizuku.rikka.app/guide/setup/#start-via-wireless-debugging-start-by-connecting-to-a-computer-shizuku-randomly-stops

**Comment by RikkaW** [comment]
Btw, if Shizuku is stopped, there is no way without user action to restart it.

**Reply by e-t-l** [reply]
Why not? It can be started on boot without user interaction. it does this already with root, or there are simple scripts/tasker/etc that do it for non-root.

I have followed all the steps to prevent the service from being killed yet I still find that Shizuku randomly has stopped - and I often have no way of knowing when this happens, except for when I try to use an app that relies on Shizuku and it throws an error. And I do suspect that it's a network change that often causes Shizuku to stop.

Is it not possible for Shizuku to create two services? The first is the regular Shizuku process, and the second would be a Helper service that simply checks if the first service is running, and either restarts it, or AT LEAST displays a notification prompting the user to restart it.

Because a big part of the problem is that for users who experience these frequent Shizuku crashes, we don't even know WHEN exactly the crash occurs.

**Reply by e-t-l** [reply]
Actually @RikkaW it looks like someone has already forked Shizuku to create such a service: https://github.com/pixincreate/Shizuku/commit/d1baee3a051a2423cbe05c9173719a4423566907#diff-b0a6d2c6850ff9eabc1ecf9b9d091928aac9de7c06ef149eca67560bf36eb9d9

Have you looked at that code? Would you be willing to merge it?
